src/pvrclient-nextpvr.cpp

```diff
@@ -416,6 +353,100 @@ void cPVRClientNextPVR::Disconnect()
   m_bConnected = false;
 }
 
+void cPVRClientNextPVR::SetVersionSpecificSettings()
+{
+  char buffer[1024];
+  //NextPVR::m_backEnd->Discovery();
+  g_livestreamingmethod = DEFAULT_LIVE_STREAM;
+
+  if (XBMC->GetSetting("livestreamingmethod", &g_livestreamingmethod))
+  {
+    // has v4 setting
+    if (m_backendVersion < 50000)
+    {
+      // previous Matrix clients had a transcoding option
+      if (g_livestreamingmethod == eStreamingMethod::Transcoded)
+      {
+        g_livestreamingmethod = eStreamingMethod::RealTime;
+        XBMC->QueueNotification(QUEUE_ERROR, XBMC->GetLocalizedString(30051), "5");
+      }
+    }
+    else  if (m_backendVersion < 50002)
+    {
+      g_livestreamingmethod = eStreamingMethod::RealTime;
+      XBMC->QueueNotification(QUEUE_ERROR, XBMC->GetLocalizedString(30051), "5.0.2+");
+    }
+    else
+    {
+      // check for new v5 setting
+      if (!XBMC->GetSetting("livestreamingmethod5", &g_livestreamingmethod))
+        if (g_livestreamingmethod == RollingFile)
+          g_livestreamingmethod = eStreamingMethod::Timeshift;
+    }
+  }
+  if (g_livestreamingmethod != eStreamingMethod::RealTime)
+  {
+    if (m_backendVersion >= 50000)
+    {
+      delete m_timeshiftBuffer;
+      if (g_livestreamingmethod == eStreamingMethod::Transcoded)
+      {
+        m_timeshiftBuffer = new timeshift::TranscodedBuffer();
+      }
+      else
+      {
+        m_supportsLiveTimeshift = true;
+        g_livestreamingmethod = eStreamingMethod::ClientTimeshift;
+        XBMC->Log(LOG_NOTICE, "Client Timeshift Based Buffering");
+        m_timeshiftBuffer = new timeshift::ClientTimeShift();
+      }
+    }
+    else
+    {
+      delete m_timeshiftBuffer;
+      m_supportsLiveTimeshift = true;
+      if (g_livestreamingmethod != eStreamingMethod::Timeshift)
+      {
+        m_timeshiftBuffer = new timeshift::RollingFile();
+      }
+      else
+      {
+        XBMC->Log(LOG_NOTICE, "Timeshift is true");
+        m_timeshiftBuffer = new timeshift::TimeshiftBuffer();
+      }
+    }
+  }
+
+  if (m_backendVersion >= 50000)
+  {
+    g_sendSidWithMetadata = false;
+    bool remote;
+    if (g_szPin != "0000" && XBMC->GetSetting("remoteaccess", &remote))
+    {
+      if (remote)
+      {
+        g_bDownloadGuideArtwork = false;
+        g_sendSidWithMetadata = true;
+      }
+    }
+  }
+  else
+  {
+    g_sendSidWithMetadata = true;
+  }
+
+  bool liveStreams;
+  if (XBMC->GetSetting("uselivestreams", &liveStreams))
+    if (liveStreams)
+      LoadLiveStreams();
```

Comment:
Maybe use brackets for the outer `if` so the scope if clear.

Reply:
Makes sense.  Changed